### PR TITLE
Improve CWL Parsing and Reconstruction

### PR DESCRIPTION
**Pull request by waleedev01**
This pull request fixes multiple issues related to CWL parsing, database storage, and CWL reconstruction. The following changes have been implemented to enhance the system's functionality and align with CWL specifications:
Fixes Implemented:
1. Fix DB to CWL Reconstruction Module (JIRA: CWL-1236):
    * Refactored the reconstruction module to accurately convert database entries into CWL files.
    * Implemented logic to exclude empty or null fields during the reconstruction process.
    * Improved the _add_inputs, _add_outputs, and _add_environment functions to ensure accurate and complete CWL reconstruction from database data.
2. Fix CWL Parsing for Output Fields:
    * Updated the CWL parsing logic to correctly handle and parse outputs fields.
    * Ensured that all output specifications, including stdout, stdin, , are accurately represented and stored in the database.
    * Improved the _add_outputs method to ensure correct parsing of output fields

### CODE REVIEW

**Comment by TaiyuGao** [comment]
It has conflicts?

**Comment by waleedev01** [comment]
> It has conflicts?

yes, will resolve them

**Comment by anisahn09** [comment]
can you also help remove `success_code`, `temporary_fail_codes`, `permanent_fail_codes` from this file? @waleedev01  https://github.com/waleedev01/analytics_automated_v2/blob/28239c50e53226462da5f27431d989c31690c073/analytics_automated/migrations/0001_initial.py#L107

File: analytics_automated/cwl_utils/cwl_clt_handler.py
Comment (waleedev01) [inline analytics_automated/cwl_utils/cwl_clt_handler.py:1]: are the output parsing changes correct here, or they are not needed? @Cenxn

Comment (Cenxn) [reply analytics_automated/cwl_utils/cwl_clt_handler.py:1]: Our previous method to extract `stdout` is actually similar to current one. No metter what the type of the output is, the [stdout](https://www.commonwl.org/v1.2/CommandLineTool.html#stdout) should specifiy the output. 

Btw, we donot use it but need we still contains `stdin, stderr`? (Just like SecondaryFiles?)

**Pull request merged**